sos/plugins/ovn_central.py

```diff
@@ -0,0 +1,122 @@
+# Copyright (C) 2018 Mark Michelson <mmichels@redhat.com>
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+from sos.plugins import Plugin, RedHatPlugin, DebianPlugin, UbuntuPlugin
+import json
+import os
+
+
+class OVNCentral(Plugin):
+    """ OVN Northd
+    """
+    plugin_name = "ovn_central"
+    profiles = ('network', 'virt')
+    option_list = [
+        ('log_dir', 'Directory where OVN logs are found', '', ''),
+        ('db_dir', 'Directory where OVN databases are found', '', ''),
+        ('runtime_dir', 'Directory where OVN runtime files are found', '', ''),
+        ('schema_dir', 'Directory where OVN database schema files are found',
+         '', '/usr/share/openvswitch/'),
+    ]
+
+    def get_dir(self, plugin_var, *candidates):
+        # If the user specified a plugin option, definitely
+        # use that value.
+        our_dir = self.get_option(plugin_var)
+        if our_dir:
+            return our_dir
+
+        for candidate in candidates:
+            if candidate and os.path.exists(candidate):
+                return candidate
```

Comment:
Does this mean just the very first candidate log directory will be collected? Dont you want to collect all of them? (and why it could be an issue to collect all of them? why dont simply call:

    self.add_copy_spec([log_dir,
                                 os.environ.get('OVS_LOGDIR'),  # works well even if the var isnt set and the cmd returns None
                                 '/var/log/containers/openvswitch',
                                 '/var/log/openvswitch',
                                 '/usr/local/var/log/openvswitch'
                                ])